### PR TITLE
[1324] Fix course status in courses show page

### PR DESCRIPTION
**Pull request by tvararu**
Replicate the logic in manage-courses-backend, taking into account the course ucas status.

### CODE REVIEW

**Comment by tvararu** [comment]
[Course says it's both not running and running](https://trello.com/c/yvzH0pPr/1324-course-says-its-both-not-running-and-running)

File: ManageCoursesUi.Tests/ViewModelHelperTests.cs [hunk 1]
```diff
@@ -52,15 +52,15 @@ class ViewModelHelperTests
...
             var schools = new List<CourseSite>
             {
-                new CourseSite {Status = status1},
-                new CourseSite {Status = status2},
-                new CourseSite {Status = status3},
-                new CourseSite {Status = status4}
+                new CourseSite {Publish = "Y", Status = status1},
+                new CourseSite {Publish = "Y", Status = status2},
+                new CourseSite {Publish = "Y", Status = status3},
+                new CourseSite {Publish = "Y", Status = status4}
```
Comment (fofr) [inline ManageCoursesUi.Tests/ViewModelHelperTests.cs]: Does there need to be a test case where Publish is set to N?

Comment (tvararu) [reply ManageCoursesUi.Tests/ViewModelHelperTests.cs]: To need, or not to need... 😄

I briefly considered doing this, but given that it's on the part of the stack that's going away, did not spend the requisite effort.

We're not testing that bit, but then again, I don't think we need to...

**Comment by misaka** [comment]
[Course says it's both not running and running](https://trello.com/c/yvzH0pPr/1324-course-says-its-both-not-running-and-running)

File: ManageCoursesUi.Tests/ViewModelHelperTests.cs (hunk 1 above)
Comment (misaka) [reply ManageCoursesUi.Tests/ViewModelHelperTests.cs]: Couldn't we just add another column to the test cases here for the Publish value?

I know we're getting rid of this "soon", but if it's fast to do I'd be tempted to do it as proof that whatever is going out here, now, actually works.

File: src/ui/Helpers/ViewModelHelpers.cs [hunk 2]
```diff
@@ -37,15 +38,19 @@ public static string GetApplicationStatus(this Course course)
         public static string GetCourseStatus(this Course course)
         {
             var result = "";
-            if (course.CourseSites.Any(s => String.Equals(s.Status, "r", StringComparison.InvariantCultureIgnoreCase)))
+            var statusRunningAndPublishedOnUcas = course.CourseSites.Any(s =>
```
Comment (defong) [inline src/ui/Helpers/ViewModelHelpers.cs]: Can probably make it look samey
``` ruby
  def ucas_status
    return :running if findable? # (RY)
    return :new if new? # (N)

    :not_running # (?)
  end```

```csharp
if(statusRunningAndPublishedOnUcas) {
return Running;
} else if (course.CourseSites.Any(s => String.Equals(s.Status, "n", StringComparison.InvariantCultureIgnoreCase)))
return NewNotRunning;

return NotRunning;
```

File: ManageCoursesUi.Tests/ViewModelHelperTests.cs (hunk 1 above)
Comment (timabell) [reply ManageCoursesUi.Tests/ViewModelHelperTests.cs]: I'll see if I can do this quickly

Comment (defong) [reply ManageCoursesUi.Tests/ViewModelHelperTests.cs]: Maybe just for R and N that's about it

Clone and amend the test and target those test case to, this test is large enough already

Comment (timabell) [reply ManageCoursesUi.Tests/ViewModelHelperTests.cs]: done, pushed to this branch

File: src/ui/Helpers/ViewModelHelpers.cs (hunk 2 above)
Comment (timabell) [reply src/ui/Helpers/ViewModelHelpers.cs]: @defong I don't know what you mean, could you suggest an improvement to the PR?

Comment (defong) [reply src/ui/Helpers/ViewModelHelpers.cs]: I meant make the ruby and csharp code syntagmatically the same looksy
```csharp
public static string GetCourseStatus(this Course course)
{
  var statusRunningAndPublishedOnUcas = course.CourseSites.Any(s =>  String.Equals(s.Status, "r", StringComparison.InvariantCultureIgnoreCase) &&
                String.Equals(s.Publish, "y", StringComparison.InvariantCultureIgnoreCase)
            );
  if(statusRunningAndPublishedOnUcas) {
    return Running;
 } else {
   if (course.CourseSites.Any(s => String.Equals(s.Status, "n",  StringComparison.InvariantCultureIgnoreCase)))
  {
   return NewNotRunning;  
  }
}
return NotRunning;
}
```

**Pull request merged**